modyn/config/schema/pipeline/trigger/cost/cost.py

```diff
@@ -0,0 +1,74 @@
+"""Cost aware triggers are evaluate the trade-off between the cost of a trigger
+(e.g. measured by training machine time) and the benefit of the trigger (e.g.
+measured by a regret metric like data incorporation latency or avoidable
+misclassification incorporation latency)."""
+
+from typing import Literal
+
+from pydantic import Field
+
+from modyn.config.schema.base_model import ModynBaseModel
+
+from ..performance.performance import _InternalPerformanceTriggerConfig
+
+
+class _CostTriggerConfig(ModynBaseModel):
+    """Base class for cost aware trigger policies."""
+
+    cost_tracking_window_size: int = Field(
+        description="How many trigger into the past should be considered for the linear cost w.r.t. data amount model."
+    )
+
+
+class DataIncorporationLatencyCostTriggerConfig(_CostTriggerConfig):
+    """Cost aware trigger policy configuration that uses the data incorporation
+    latency as a regret metric.
+
+    While no trigger occurs the un-integrated data is cumulated and the sum of all seconds of every unincorporated data
+    point is added up.
+
+    `incorporation_delay_per_training_second` servers as a conversion rate between budget (training time) and regret
+    metric (incorporation latency).
+
+    When the regret metric exceeds the training-time based budget metric, a trigger is fired.
```

Comment:
Sorry, I might just be remembering wrongly - a cost/accuracy tradeoff is not happening here, only time based? Is that exactly what you were trying to tell me? Maybe we should also document this., if that is true.

Reply:
`DataIncorporationLatencyCostTriggerConfig` is only cost-aware but not performance-aware. 
The policy is inspired by https://cse.hkust.edu.hk/~weiwa/papers/continuum-socc18.pdf (p. 7; 4.3).

It's time-sensitive and sensitive to data/query density. If there are e.g. periods where almost no data comes in,  triggers are avoided. In some sense, this is a combination of time- and amount-based approaches. 

This isn't too beneficial for our evaluation setting as we weight our evaluations by time and not by data amount.

I expect more from the second cost policy that includes the misclassifiations.